### PR TITLE
Clarify token metadata and relationship to RFC1464

### LINKED ISSUE #94: Clarifications relative to RFC1464 (including need to better specify key=value encoding)
We reference and use RFC1464 in the metadata sub-section but unless I'm misreading there inconsistency here, and the resulting syntax doesn't match RFC1464 and is under-specified and has some ambiguities.  

In particular:

* RFC1464 doesn't actually define the comma-separated scheme we specify. Unless I'm misreading it, the "token=XYZ,expiry=ABC" would just result in "XYZ,expiry=ABC" being the value of "token".  There seem to be lots of different approaches here (eg, DNS-SD / RFC6763 has them as separate length/value units while DKIM uses ";"as a separator, and I suspect other approaches exist).
* We first recommend just using the base64uri-encoded token, but in the following metadata section we recommend using token=.  Should we be consistent and always recommend a RFC1464 token= mechanism?
* As base64uri is often going to have trailing "=" as padding, we may want to include this in examples and make sure this is unambiguous as far as escaping/encoding goes.  Otherwise this could be a common interop issue.

### PR DESCRIPTION
**Pull request by enygren**
Also includes the validation record -> Validation Record changes
Fixes #94 and includes changes from #132

### CODE REVIEW

File: draft-ietf-dnsop-domain-verification-techniques.md [hunk 1]
```diff
@@ -539,3 +556,4 @@ While the ACME protocol ([RFC8555]) specifies a way to demonstrate ownership ove
 # Acknowledgments
 
 Thank you to Tim Wicinski, John Levine, Daniel Kahn Gillmor, Amir Omidi, Tuomo Soini, and many others for their feedback and suggestions on this document.
+
```
Comment (enygren) [inline draft-ietf-dnsop-domain-verification-techniques.md:559]:
```suggestion
```

File: draft-ietf-dnsop-domain-verification-techniques.md [hunk 2]
```diff
@@ -365,23 +382,23 @@ Application Service Providers may wish to always prepend the `_<identifier-token
```
Comment (enygren) [inline draft-ietf-dnsop-domain-verification-techniques.md:382]: We should remove the "(TODO / DISCUSS: should we remove the underscore in-front of identifier-token per the guidance above?)"

But to that point, do we update the above example to not have the underscore before "_<identifier-token>" or do we change the proposed guidance?

File: draft-ietf-dnsop-domain-verification-techniques.md [hunk 3]
```diff
@@ -248,17 +248,18 @@ A unique token used in the challenge. It should be a random value issued between
...
-3. MUST NOT start with an underscore to avoid attacks where tokens are maliciously used as a scope label.
 
 See {{RFC4086}} for additional information on randomness requirements.
 
 Base32 encoding or hexadecimal base16 encoding are RECOMMENDED to be specified when the random token would exist in a DNS label such as in a CNAME target.  This is because base64 relies mixed case (and DNS is case-insensitive as clarified in {{RFC4343}}) and because some base64 characters ("/", "+", and "=") may not be permitted by implementations that limit allowed characters to those allowed in hostnames.  If base32 is used, it SHOULD be specified in way that safely omits the trailing padding ("=").  Note that DNS labels are limited to 63 octets which limits how large such a token may be.
 
 This random token is placed in either the RDATA or a domain name, as described in the rest of this section.  Some methods of validation may involve multiple independent random tokens.
 
+When placed in a domain name, the token MUST NOT start with an underscore to avoid attacks where tokens are maliciously used as a scope label.
```
Comment (enygren) [inline draft-ietf-dnsop-domain-verification-techniques.md:258]: This (which was the same text but in bullet 3 in the previous version) is in-conflict with some of our examples.

File: draft-ietf-dnsop-domain-verification-techniques.md (hunk 2 above)
Comment (enygren) [reply draft-ietf-dnsop-domain-verification-techniques.md:382]: FWIW, draft-ietf-acme-scoped-dns-challenges-00 does include an underscore here.
eg, in its example:
`_ujmmovf2vn55tgye._acme-wildcard-challenge.example.org 300 IN TXT "LoqXcYV8...jxAjEuX0.9jg46WB3...fm21mqTI"`

File: draft-ietf-dnsop-domain-verification-techniques.md [hunk 4]
```diff
@@ -280,37 +281,54 @@ Application owners SHOULD consult the IANA "Underscored and Globally Scoped DNS
...
 Application Service Providers MUST validate that a random token in the TXT record matches the one that they gave to the User for that specific domain name.
 
-### Metadata For Expiry {#metadata}
+### Token Metadata {#metadata}
+
+It may be desirable to associate metadata with the token in a Validation Record. When specified, metadata SHOULD be encoded in the RDATA via space-separated ASCII key-value pairs {{RFC1464}}, with the key "token" prefixing the random token. For example:
+
+    _foo-challenge.example.com.  IN   TXT  "token=3419...3d206c4"
+
+If there are multiple tokens required, each one MUST be in a seperate RR to allow them to match up with any additional attributes.  For example:
```
Comment (ShivanKaul) [inline draft-ietf-dnsop-domain-verification-techniques.md:306]:
```suggestion
If there are multiple tokens required, each one MUST be in a separate RR to allow them to match up with any additional attributes.  For example:
```

File: draft-ietf-dnsop-domain-verification-techniques.md [hunk 5]
```diff
@@ -280,37 +281,54 @@ Application owners SHOULD consult the IANA "Underscored and Globally Scoped DNS
...
 
     _foo-challenge.example.com.  IN   TXT  "3419...3d206c4"
 
-This again allows the Application Service Provider to query only for application-specific records it needs, while giving flexibility to the User adding the DNS record (i.e. they can be given permission to only add records under a specific prefix by the DNS administrator). Whether or not multiple validation records can exist for the same domain is up to the Application Service Provider's application specification.
+This again allows the Application Service Provider to query only for application-specific records it needs, while giving flexibility to the User adding the DNS record (i.e. they can be given permission to only add records under a specific prefix by the DNS administrator). Whether or not multiple Validation Records can exist for the same domain is up to the Application Service Provider's application specification.
 
 Application Service Providers MUST validate that a random token in the TXT record matches the one that they gave to the User for that specific domain name.
 
-### Metadata For Expiry {#metadata}
+### Token Metadata {#metadata}
```
Comment (ShivanKaul) [inline draft-ietf-dnsop-domain-verification-techniques.md:300]: (note for reviewers: this is the main part of this PR to review)

**Pull request merged**